parser.py

```diff
@@ -38,14 +38,16 @@
 # Print a compile-time error in Python < 3.6. This line does nothing in Python 3.6+ but is reported to the user
 # as an error (because it is the first line that fails to compile) in older versions.
 f' Error: This script requires Python 3.6 or later. Use `python --version` to check your version.'
-
+# To do it more explicitly:
+assert sys.version_info >= (3, 6)
```

Comment:
This assertion can never be hit. Python gets compiled to byte-code before being executed. If you try to run the script on anything less than 3.6 then the script won't even be executed due to the compile-time syntax errors, like f-strings.